src/config.rs

```diff
@@ -0,0 +1,44 @@
+use alloc::string::String;
+use alloc::sync::Arc;
+use alloc::vec::Vec;
+use axdevice_base::BaseDeviceOps;
+use axerrno::AxResult;
+
+//！Represents the configuration of an emulated device for a virtual machine.
+#[derive(Debug, Default, Clone, serde::Serialize, serde::Deserialize)]
+pub struct EmulatedDeviceConfig {
+    pub name: String,
+    /// The base IPA (Intermediate Physical Address) of the device.
+    pub base_ipa: usize,
+    /// The length of the device.
+    pub length: usize,
+    /// The IRQ (Interrupt Request) ID of the device.
+    pub irq_id: usize,
+    /// The type of emulated device.
+    pub emu_type: usize,
+    pub cfg_list: Vec<usize>,
+}
+
+/// Implementation of methods for EmuDeviceType.
+impl EmulatedDeviceConfig {
+    pub fn to_emu_dev(&self) -> AxResult<Arc<dyn BaseDeviceOps>> {
```

Comment:
It looks so strange to impl this function to `EmulatedDeviceConfig`

Reply:
it is used for devices init

Reply:
Such constructor should be implemented by xxx_device, through method like 
```Rust
impl VirtioBlock {
    fn init(config: EmulatedDeviceConfig) -> Self {}
}
```